graphjet-core/src/main/java/com/twitter/graphjet/algorithms/counting/tweet/TopSecondDegreeByCountForTweet.java

```diff
@@ -55,9 +56,51 @@ public TopSecondDegreeByCountForTweet(
     super(leftIndexedBipartiteGraph, expectedNodesToHit, statsReceiver);
   }
 
+  protected long getEdgeTimeStampInMillis(
+    long timeStampFromTweetId,
+    byte edgeType,
+    long edgeMetadata
+  ) {
+    switch (edgeType) {
+      case 0:  // CLICK
+        return edgeMetadata;
+      case 1:  // FAVORITE
+        return edgeMetadata;
+      case 2:  // RETWEET
+        return RecentTweetFilter.timeStampFromTweetId(edgeMetadata);
```

Comment:
Just to make sure:

1) For favorites, the edge timestamp is the timestamp of the Favorite.
2) For retweets, replies etc, the edge timestamp is the timestamp of the original tweet
3) Why do we treat isMentioned and isMediatagged separately? Can't we just use RecentTweetFilter.timeStampFromTweetId for consistency?